src/test/scala/ca/ferlab/ferload/client/configurations/UserConfigTest.scala

```diff
@@ -0,0 +1,45 @@
+package ca.ferlab.ferload.client.configurations
+
+import org.scalatest.BeforeAndAfter
+import org.scalatest.funsuite.AnyFunSuite
+
+import java.io.FileInputStream
+import java.util.Properties
+
+class UserConfigTest extends AnyFunSuite with BeforeAndAfter {
+
+  val path: String = "/tmp/.ferload-client.properties"
+  val userConfig = new UserConfig(path)
+
+  before {
+    userConfig.clear()
+  }
+
+  test("set / get") {
+    userConfig.set(Username, "foo")
+    assert(userConfig.get(Username).equals("foo"))
+  }
+
+  test("clear") {
+    userConfig.set(Username, "foo")
+    assert(userConfig.get(Username).equals("foo"))
+    userConfig.clear()
+    assert(userConfig.get(Username) == null)
+  }
+
+  test("save") {
+    assert(getCurrentSavedValue(Username) == null)
+    userConfig.set(Username, "foo")
+    userConfig.save()
+    assert(userConfig.get(Username).equals("foo"))
+  }
+
+  def getCurrentSavedValue(userConfigName: UserConfigName): String = {
+    val props = new Properties()
+    val fis = new FileInputStream(path)
+    props.load(fis)
+    fis.close()
+    Option(props.get(userConfigName.name)).map(_.toString).orNull
```

Comment:
You could return an Option[String] instead of String which could be null
Personally, I never use null in scala, always None and Option :)